src/compiler/rust.rs

```diff
@@ -1896,6 +1896,27 @@ impl pkg::InputsPackager for RustInputsPackager {
             let dist_input_path = path_transformer.as_dist(&input_path).with_context(|| {
                 format!("unable to transform input path {}", input_path.display())
             })?;
+
+            if input_path.ends_with("src/lib.rs") {
```

Comment:
Should it works on Windows too?

Reply:
no idea where `sccache-dist` is supported

Reply:
I guess building `"src/lib.rs"` using `PathBuf::new().join("src").join("lib.rs")` should be enough, if this is separators you are worrried about?

Reply:
I'm not sure in fact we need ?
```
PS C:\Users\user> rustc .\test.rs
PS C:\Users\user> cat .\test.rs
use std::path::PathBuf;

fn main() {
    let p = PathBuf::new().join("src").join("lib.rs");
    println!("p={:?}", p);
    let crate_path = PathBuf::new().join("crate1/src/lib.rs");
    println!("path => {:?}", crate_path.ends_with(p));
}
PS C:\Users\user> .\test.exe
p="src\\lib.rs"
path => true
PS C:\Users\user>

Reply:
I prefer to rely on cross-platfom solution if it's possible

Reply:
What do you have in mind as cross-platform ? using `PathBuf::new().join("src").join("lib.rs");` or something else?

Reply:
Yes, I suggest to use `PathBuf::new().join("src").join("lib.rs");`

Reply:
should be good now